src/Gd/Imagine.php

```diff
@@ -208,9 +213,14 @@ private function requireGdVersion($version)
      */
     private function doLoad($string, MetadataBag $metadata)
     {
-        $resource = $this->withoutExceptionHandlers(function () use (&$string) {
-            return @imagecreatefromstring($string);
-        });
+
+        if($this->isWebp($string)) {
```

Comment:
`imagecreatefromwebp` has been introduced in PHP 5.4, but Imagine [supports PHP 5.3.2](https://github.com/avalanche123/Imagine/blob/1.2.0/composer.json#L20). so what about

```suggestion
        if(function_exists('imagecreatefromwebp') && $this->isWebp($data)) {
```

Reply:
Well, you have to choose whether you want to support version that is 10 years old and unsupported since last 5 years or to use the image standard from 2010, early adopted by facebook & google in 2013 :) I'm all up for the latter but I can work on my fork so it's up to you.

Reply:
Well, you have to choose whether you want to support version that is 10 years old and unsupported since last 5 years or to use the image standard from 2010, early adopted by facebook & google in 2013 :) I'm all up for the latter but I can work on my fork so it's up to you.

Reply:
Well, you have to choose whether you want to support version that is 10 years old and unsupported since last 5 years or to use the image standard from 2010, early adopted by facebook & google in 2013 :) I'm all up for the latter but I can work on my fork so it's up to you.

Do you think that function check is good enough? Shouldn't we post some information that with the newer version of PHP this will be working at least?

(Sorry for the other comment, wrong account).

Reply:
I'd just do the function exists check. We can add a note to the changelog / release notes that webp support requires PHP 5.4+.

Reply:
I've pushed the new commit with your suggestion in it :)